compatibility/common/src/main/java/org/projectnessie/tools/compatibility/internal/OldNessieServer.java

```diff
@@ -91,13 +94,16 @@ static ClassLoader createClassLoader(Version version) {
     // The 'nessie-jaxrs' has all the necessary dependencies to the DatabaseAdapter
     // implementations, REST services, Version store implementation, etc. in older versions.
     // Newer versions declare what is required as runtime dependencies of
-    // `nessie-compatibility-common`
-    String artifactId =
+    // `nessie-compatibility-common`, except mongodb
+    List<String> artifactIds =
         version.isGreaterThanOrEqual(COMPAT_COMMON_DEPENDENCIES_START)
-            ? "nessie-compatibility-common"
-            : "nessie-jaxrs";
+            ? asList(
+                "nessie-compatibility-common",
+                "nessie-versioned-persist-mongodb",
```

Comment:
The mongodb dependencies artifacts are not present in `nessie-compatiblity-common` (and not necessary there)